### ISSUE TITLE
vendor and node_module still exist in time machine

### ISSUE THREAD

**Issue by tanthammar** [stevegrunwell/asimov#45]
Apologies for asking, but I don't understand how any of this works, wandering in a black hole ...

1. When I launch Time Machine all my projects, vendor and node_modules are still there.
2. When I run `sudo mdfind "com_apple_backup_excludeItem = 'com.apple.backupd'` none of the vendor/node_modules, dirs are listed.

I launched the daemon a few days ago and thought that the newer backups wouldn't include those folders?

- I followed the installation instructions. 
   - Got "Asimov daemon has been loaded!" in terminal
   - In the list generated from the process, **I could see my projects in the "Sites" dir being added for exclusion.**

My "web" projects are either in the "Sites" folder or in my "Github" folder on my mac.
Each of them have node_modules and vendor folders ...

**Comment by codexp** [comment]
Once backup has been done for those folders, they will remain for ever.
Excluding a directory from backup will merely prevent time machine backing up newer changes, but old files will persist.

You can manually delete those folders from time machine for all backups.

Here is a link how to do that:
http://osxdaily.com/2015/07/27/delete-old-backups-time-machine-mac/

**Comment by stevegrunwell** [comment]
As @codexp mentioned above, Asimov doesn't remove those directories from existing backups, it only prevents them from being added to or updated in Time Machine.

This does raise a gap in the documentation, however, so I'm going to keep the issue open. It might also be worth considering a "wipe these directories out from backups"-type command to purge dependencies that have already made it to Time Machine 🤔

**Comment by ajoslin103** [comment]
When time machine creates a new backup - are the exclusions then fully in force?

IOW: If TM creates a new backup does that reset everything and there is no longer a need for a "wipe these directories out from backups"-type command ?

**Comment by j-f1** [comment]
I don’t think so. Time Machine simply won’t add the updated directories to the backup, but they’ll continue to exist in old backups.

**Comment by terryupton** [comment]
At what point then would these folders be listed in the `sudo mdfind "com_apple_backup_excludeItem = 'com.apple.backupd'` command? 

Does this only happen if they are never added from the outset?

**Comment by django23** [comment]
@tanthammar @terryupton @ajoslin103 — answering the thread properly, better late than never. 🙏

@codexp had it right: **Asimov never touches existing backups.** An exclusion stops Time Machine backing a directory up *going forward*; copies already on the backup disk stay until you delete them by hand or Time Machine ages them out.

To check what's actually excluded now:

```sh
sudo mdfind "com_apple_backup_excludeItem = 'com.apple.backupd'"
asimov --dry-run    # shows what Asimov would exclude, changes nothing
```

If `mdfind` shows nothing, the run isn't reaching your projects. Two common causes: projects outside `$HOME` (fixed with `[scan] extra`, see the README) and the schedule not being loaded (`launchctl print gui/$(id -u)/com.stevegrunwell.asimov`).

Keeping this open until the README says all of that plainly — "why is it still in my backup?" comes up often enough that it belongs in the docs, not an issue thread. The purge-old-backups idea is tracked in #38.

**Comment by django23** [comment]
The README now says this outright, so closing the loop here. 🎉

The new [**What Asimov doesn't do**](https://github.com/AsimovMac/asimov#what-asimov-doesnt-do) section states plainly that an exclusion is forward-looking — copies already on the backup disk stay there — and adds a short recipe for checking whether a run is actually reaching your projects:

```sh
asimov --dry-run                                                   # what Asimov would exclude
sudo mdfind "com_apple_backup_excludeItem = 'com.apple.backupd'"   # what's actually excluded
```

Thanks @tanthammar for asking the question that turned out to be the most common one in the tracker, and @codexp for answering it correctly six years before the docs did. 🙏

Closing as documented. Removing stale entries from Asimov's own list is tracked in #38.
